src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -45,9 +45,52 @@ PitchShiftGroupState::PitchShiftGroupState(
     initializeBuffer(engineParameters);
     audioParametersChanged(engineParameters);
 
+    // Fill the buffer with zero values.
     SampleUtil::clear(m_retrieveBuffer[0], kPrefilledFrames);
     SampleUtil::clear(m_retrieveBuffer[1], kPrefilledFrames);
 
+    // Prefilling the RubberBand with zero samples. Based on filling the inner
+    // RubberBand buffers with samples, the total latency is lower. It is caused
+    // due to without filling the inner buffers, it will take more input
+    // sample buffers to produce the "meaningful" output. If some number
+    // of input frames are provided into RubberBand, we can't expect
+    // the same number of frames on the output, especially at the start
+    // of RubberBand processing. To dive a little more detailed
+    // into this problem, when the number of frames of the input buffer is not
+    // sufficient for RubberBand, the RubberBand will not produce enough
+    // output frames to fill the output buffer or actually any frames at all.
+    // At first, the RubberBand will not produce result frames. Then,
+    // it will produce some frames, but not enough to fill the whole
+    // output buffer with the same amount of frames as in the input buffer.
+    // It will take more input buffers until the RubberBand will produce enough
+    // frames to fill the entire output buffer. This situation can be shown
+    // in the following diagram:
+    //
+    // The X letter represents the real frame data.
+    //
+    // stream of input buffers:
+    //
+    // |XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
+    //
+    // stream of output buffers:
+    //
+    // |        |        |        |    XXXX|    XXXX|  XXXXXX|XXXXXXXX|XXXXXXXX| ...
+
+    // By using the RubberBand inner buffers prefilling with zero samples (silence),
+    // the latency will be exactly the same as the specific optimal amount
+    // of offered zero samples. The optimal amount of prefilled sample frames
+    // represent the kPrefilledFrames constant. The situation can be shown
+    // in the following diagram:
+    //
+    // The X letter represents the real frame data.
+    //
+    // stream of input buffers:
+    //
+    // |00000000|00000000|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
+    //
+    // stream of output buffers:
+    //
+    //                   |00000000|00000000|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
```

Comment:
What is the difference between 0 and X inside Rubberband? Isn't 0 just a special X? Probably the visual is just misleading.
Is each bar a callback? I think than you may visualize this: 

```
without zero initalization 
in:    |XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
out:   |        |        |        |    XXXX|    XXXX|  XXXXXX|XXXXXXXX|XXXXXXXX| ...

6 x out broken, if we skip 5 cycles we have enough samples in the 6th,   

with two cycles zero initialization 
in:    |00000000|00000000|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
out:   |        |        |        |        |        |00000000|00000000|XXXXXXXX|XXXXXXXX| ...
 

with zero initalization during init 
init:  |00000000|00000000|00000000|00000000|00000000|
in:    |XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
out:   |00000000|00000000|00000000|00000000|00000000|XXXXXXXX| ...

Reply:
My idea is that we can do the same wit a start delay and real samples: 

```
without zero initalization 
in:    |XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
out:   |        |        |        |    XXXX|    XXXX|  XXXXXX|XXXXXXXX|XXXXXXXX| ...

6 x out broken, if we skip 5 cycles we have enough samples in the 6th,  
Fill output with 0 during the first 5 cycles  

in:    |XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
out:   |00000000|00000000|00000000|00000000|00000000|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...

since de group delay is not const put some X into a ring and comsume it whenever Rubberband request it via getSamplesRequired() 
 
in:    |XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX|XXXXXXXX| ...
out:   |00000000|00000000|00000000|00000000|00000000|00000000|XXXXXXXX|XXXXXXXX| ...

```